scripts/tests/test_validate_docstrings.py

```diff
@@ -229,6 +229,24 @@ def good_imports(self):
         """
         pass
 
+    def does_nothing(self, x):
+        """
+        Do nothing and always return none.
+
+        But in other cases it might also
+        return nothing at all.
+
+        Parameters
+        ----------
+        x : int
+            Useless parameter.
+        """
+        var_with_return_on_the_name = None
+        if var_with_return_on_the_name:
```

Comment:
Any reason to not have a `if True` or `if False` and define that variable?

Also, please remove the comments in the next 3 lines, they don't add value anymore.

Reply:
Please, read the following as a curiosity, and not an argument.

The variable and the comments represented cases where the regex approach could fail. By then, my reasoning went like: if someone try to simplify this regex in the future, the tests should raise errors.

Now, using the AST, comments and variable names are irrelevant. Nonetheless, if someday, by whatever reason, someone updates the code to a different approach, be it again with regex or something else, shouldn't the already existing tests act as a prevention against new, unexpected flaws?

I'd just like to understand a little further the principles behind designing tests from the perspective of someone experienced. I'll me making the suggested changes either way.

Reply:
I have updated and minimized the tests to better reflect the guidelines you gave me.